projects/storefrontlib/src/lib/product/components/product-details/product-summary/product-summary.component.html

```diff
@@ -1,24 +1,42 @@
-<div class="cx-product-summary">
-  <div class="cx-product-summary__price cx-product-summary__price" aria-label="new item price">
-    {{product?.price?.formattedValue}}
-  </div>
-
-  <div class="cx-product-summary__description">
-    <p [innerHTML]="product?.summary"> </p>
-  </div>
+<div class="product-summary">
+  <ng-container *cxOutlet="outlets.TITLE">
+    <div class="item-title">
+      <div class="name">{{ product?.name }}</div>
+      <div class="code">ID {{ product?.code }}</div>
+    </div>
+  </ng-container>
 
-  <div class="cx-product-summary__variant">
-    <div class="cx-product-summary__variant-label">
-      Qty:
+  <ng-container *cxOutlet="outlets.RATING">
+    <div class="item-rating">
+      <cx-star-rating [rating]="product?.averageRating" [disabled]="true"></cx-star-rating>
+      <div class="review-count">({{ product?.numberOfReviews }})</div>
+      <a class="review-link btn-link">Write a review</a>
     </div>
-    <cx-item-counter [min]="1" [max]="product.stock.stockLevel" *ngIf="product?.stock?.stockLevel > 0 || product?.stock?.stockLevelStatus === 'inStock'"
-      (update)="updateCount($event)"></cx-item-counter>
-    <span class="cx-product-summary__variant-info">
-      <span *ngIf="product?.stock?.stockLevel > 0 || product?.stock?.stockLevelStatus === 'inStock'">
-        {{product?.stock?.stockLevel}} in stock</span>
-      <span *ngIf="product?.stock?.stockLevel == 0">Out of stock</span>
-    </span>
-  </div>
+  </ng-container>
 
+  <div class="price" aria-label="new item price">
+    {{ product?.price?.formattedValue }}
+  </div>
+  <div class="description"><p [innerHTML]="product?.summary"></p></div>
 
-</div>
\ No newline at end of file
+  <ng-container *cxOutlet="outlets.ADDTOCART">
+    <div class="quantity">
```

Comment:
I'd flatten and simplify the whole block a bit more:

```html
<div class="quantity">
    <label>Qty:</label>

    <cx-item-counter></cx-item-counter>

    <span class="info">
      {{stockInfo}}
    </span>

    <cx-add-to-cart
      *ngIf="product?.stock?.stockLevelStatus !== 'outOfStock'"
      [quantity]="itemCount"
      [productCode]="product?.code"
    ></cx-add-to-cart>

</div>
```

**note** I also moved out the stocklevel concatenation logic, let's move it now to the control logic, soon it will go in a service so we can override it more easily.

Reply:
yes,. let's move the logic to a public method in the component for now.

Reply:
Nicely done. I've pushed a small change to the code.